core/src/serve_repair.rs

```diff
@@ -187,15 +195,46 @@ impl ServeRepair {
         blockstore: Option<&Arc<Blockstore>>,
         requests_receiver: &PacketReceiver,
         response_sender: &PacketSender,
+        stats: &mut ServeRepairStats,
     ) -> Result<()> {
         //TODO cache connections
         let timeout = Duration::new(1, 0);
         let reqs = requests_receiver.recv_timeout(timeout)?;
+        stats.total_packets += reqs.packets.len();
+        // Drop the rest in the channel in case of dos
+        while let Ok(more) = requests_receiver.try_recv() {
```

Comment:
hmmm do we want to be dropping everything after the first batch? There may be some requests that have queued up in the time it took us to run the last iteration of this function. What about some rate limit on number of repairs handled/second, and then dump after that has been reached?

Reply:
yea.. we could do something smarter here, but that's more work :) I have some ideas to try.

Reply:
Hmmm have tried this on a testnet? I'm worried this will break repair/make it slow, we could be dropping a lot of things if a batch takes a while to process.

Reply:
Haven't yet, but definitely will before going with any fix. I'm trying to fix the metrics so I can actually see what is going on without it being too spammy.

Reply:
Ok, I have a new strategy of maintaining a max number of packets around 1s and adjust by 10% based on the last speed if it is at the max. Doesn't seem like we drop any packets in the normal bench-tps run. I attached a picture.